libs/sdk-ui-dashboard/src/model/commandHandlers/filterContext/filterViewHandler.ts

```diff
@@ -0,0 +1,149 @@
+    filterViewDeletionFailed,
+    filterViewApplicationFailed,
+    filterViewDefaultStatusChangeFailed,
+    filterViewDeletionSucceeded,
+    filterViewApplicationSucceeded,
+    filterViewDefaultStatusChangeSucceeded,
+} from "../../events/filters.js";
+import { loadFilterViews } from "../dashboard/initializeDashboardHandler/loadFilterViews.js";
+import { PromiseFnReturnType } from "../../types/sagas.js";
+
+function createFilterView(
+    ctx: DashboardContext,
+    filterView: IDashboardFilterViewSaveRequest,
+): Promise<IDashboardFilterView> {
+    return ctx.backend.workspace(ctx.workspace).dashboards().createFilterView(filterView);
+}
+
+export function* saveFilterViewHandler(ctx: DashboardContext, cmd: SaveFilterView): SagaIterator<void> {
+    if (!ctx.dashboardRef) {
+        throw Error("Dashboard ref must be provided.");
+    }
+
+    const filterContext: ReturnType<typeof selectFilterContextDefinition> = yield select(
+        selectFilterContextDefinition,
+    );
+
+    const filterView: IDashboardFilterViewSaveRequest = {
+        name: cmd.payload.name,
+        dashboard: ctx.dashboardRef,
+        filterContext,
+        isDefault: cmd.payload.isDefault,
+    };
+
+    try {
+        const newFilterView: PromiseFnReturnType<typeof createFilterView> = yield call(
+            createFilterView,
+            ctx,
+            filterView,
+        );
+        yield put(reloadFilterViews());
+        yield put(filterViewCreationSucceeded(ctx, newFilterView, cmd.correlationId));
+    } catch (error) {
+        defaultErrorHandler(error);
+        yield put(filterViewCreationFailed(ctx));
+    }
+}
+
+function deleteFilterView(ctx: DashboardContext, ref: ObjRef): Promise<void> {
+    return ctx.backend.workspace(ctx.workspace).dashboards().deleteFilterView(ref);
+}
+
+export function* deleteFilterViewHandler(ctx: DashboardContext, cmd: DeleteFilterView): SagaIterator<void> {
+    try {
+        const filterView: PromiseFnReturnType<typeof findFilterView> = yield call(
+            findFilterView,
+            cmd.payload.ref,
+        );
+
+        yield call(deleteFilterView, ctx, cmd.payload.ref);
+        yield put(reloadFilterViews());
+        yield put(filterViewDeletionSucceeded(ctx, filterView, cmd.correlationId));
+    } catch (error) {
+        defaultErrorHandler(error);
+        yield put(filterViewDeletionFailed(ctx));
+    }
+}
+
+function* findFilterView(ref: ObjRef) {
+    const filterViews: ReturnType<typeof selectFilterViews> = yield select(selectFilterViews);
+    return filterViews.find((filterView) => areObjRefsEqual(filterView.ref, ref));
+}
+
+export function* applyFilterViewHandler(ctx: DashboardContext, cmd: ApplyFilterView): SagaIterator<void> {
+    const filterView: PromiseFnReturnType<typeof findFilterView> = yield call(
+        findFilterView,
+        cmd.payload.ref,
+    );
+
+    if (filterView) {
+        yield put(
+            changeFilterContextSelectionByParams({
```

Comment:
is it working with current AF config in state if it is not provided in params?

Reply:
Seems like it is probably not using it. I plan to look into this. I would leave it as is in this pull request and will fix it in the next one.

Reply:
I think I saw the usage of `changeFilterContextSelectionByParams` in reset filter feature originally. I looked into it some more and changed the command to `changeFilterContextSelection` and I'm not messing with configs any more.